components/eamxx/src/dynamics/homme/eamxx_homme_iop.cpp

```diff
@@ -333,67 +338,60 @@ apply_iop_forcing(const Real dt)
   // TODO: Create a memory buffer for this class
   //       and add the below WSM and views
   WorkspaceMgr eamxx_wsm(NLEVI, 7+qsize, policy_eamxx);
-  WorkspaceMgr homme_wsm(NLEV, 16 + (theta_hydrostatic_mode ? 16 : 0), policy_homme);
   view_Nd<Pack, 4>
-    rstar      ("rstar",       nelem, NGP, NGP, NLEV),
-    exner      ("exner",       nelem, NGP, NGP, NLEV),
     temperature("temperature", nelem, NGP, NGP, NLEV);
 
-  // Lambda for computing rstar, exner, and temperature from Hommexx
+  // Lambda for computing temperature from Hommexx
   auto compute_homme_states = [&] () {
-    Kokkos::parallel_for("compute_rstar_exner_and_temperature", policy_homme, KOKKOS_LAMBDA (const KT::MemberType& team) {
+    Kokkos::parallel_for("compute_temperature_for_iop", policy_homme, KOKKOS_LAMBDA (const KT::MemberType& team) {
       KV kv(team);
       const int ie  =  team.league_rank();
 
       // Get temp views from workspace
-      auto ws = homme_wsm.get_workspace(team);
-      auto pnh_slot   = ws.take_macro_block("pnh"  , NGP*NGP);
-      uview_2d<Pack> pnh(reinterpret_cast<Pack*>(pnh_slot.data()), NGP*NGP, NLEV);
+      auto ws = eamxx_wsm.get_workspace(team);
+      uview_1d<Pack> pmid, pint, pdel;
+      ws.take_many_contiguous_unsafe<1>({"pmid"},{&pmid});
 
       Kokkos::parallel_for(Kokkos::TeamThreadRange(kv.team, NGP*NGP), [&] (const int idx) {
         const int igp = idx/NGP;
         const int jgp = idx%NGP;
 
+        auto ps_i          = ps_dyn(ie, igp, jgp);
         auto dp3d_i        = ekat::subview(dp3d_dyn, ie, igp, jgp);
         auto vtheta_dp_i   = ekat::subview(vtheta_dp_dyn, ie, igp, jgp);
-        auto phi_int_i     = ekat::subview(phi_int_dyn, ie, igp, jgp);
         auto qv_i          = ekat::subview(Q_dyn, ie, 0, igp, jgp);
-        auto pnh_i         = ekat::subview(pnh, idx);
-        auto rstar_i       = ekat::subview(rstar, ie, igp, jgp);
-        auto exner_i       = ekat::subview(exner, ie, igp, jgp);
         auto temperature_i = ekat::subview(temperature, ie, igp, jgp);
 
+        // Compute reference pressures and layer thickness.
+        // TODO: Allow geometry data to allocate packsize
+        auto s_pmid = ekat::scalarize(pmid);
+        Kokkos::parallel_for(Kokkos::TeamVectorRange(team, total_levels), [&](const int& k) {
```

Comment:
This is related to my overall comment about these policy change below. But we should not use a `TeamVectorRange` here, because we are inside a `TeamThreadRange`. We should use a `ThreadVectorRange`.

The possible patterns for team policies are 
```
TeamPolicy {
  TeamThreadRange {
    ThreadVectorRange {
  }
}
```
or
```
TeamPolicy {
  TeamVectorRange {
}
```
But ignore this comment if you decide to change all the policies to the `policy_eamxx` where there would be no need for the `TeamThreadRange`.